akka-stream-tests/src/test/scala/akka/stream/io/TcpSpec.scala

```diff
@@ -692,6 +698,90 @@ class TcpSpec extends StreamSpec("akka.stream.materializer.subscription-timeout.
     }
   }
 
+  "TLS client and server convenience methods" should {
+
+    "allow for 'simple' TLS" in {
+      // cert is valid until 2025, so if this tests starts failing after that you need to create a new one
+      val (sslContext, firstSession) = initSslMess()
+      val address = temporaryServerAddress()
+
+      Tcp().bindAndHandleTls(
+        // just echo charactes until we reach '\n', then complete stream
+        // also - byte is our framing
+        Flow[ByteString].mapConcat(_.utf8String.toList)
```

Comment:
:) New record in allocation-olympics! :)

Reply:
🥇